slybot/slybot/plugins/scrapely_annotations/processors.py

```diff
@@ -100,13 +100,8 @@ def _process_fields(self, data):
         for field_num, (field, value) in enumerate(self._normalize_data(data)):
             # Repeated Fields and nested items support
             if hasattr(field, 'fields'):
-                child = None
-                if len(field.fields) == 1:
-                    child = field.fields.values()[0]
-                if child and field.descriptor == child.extractor:
-                    fields[child.id] = child
-                else:
-                    fields[field.name] = field
+                _id, field = self._process_item(field)
+                fields[_id] = field
```

Comment:
This would be cleaner if you do `self._add_field(field, fields)` and do the `fields[_id] = field` in the helper method.